hibernate-core/src/main/java/org/hibernate/dialect/aggregate/OracleAggregateSupport.java

```diff
@@ -138,27 +145,69 @@ public String aggregateComponentCustomReadExpression(
 										placeholder,
 										"json_value(" + parentPartExpression + columnExpression + "' returning " + column.getColumnDefinition() + ')'
 								);
+
 							case DATE:
-								return template.replace(
-										placeholder,
-										"to_date(json_value(" + parentPartExpression + columnExpression + "'),'YYYY-MM-DD')"
-								);
+								if (this.dateTypesStoreAsString) {
+									return template.replace(
+											placeholder,
+											"to_date(json_value(" + parentPartExpression + columnExpression + "'),'YYYY-MM-DD')"
```

Comment:
Since the OSON encoding seems to add `T00:00:00`, the backwards compatible change here would be to do this:

```suggestion
											"to_date(substr(json_value(" + parentPartExpression + columnExpression + "'),1,10),'YYYY-MM-DD')"
```

This is what my local testing showed:

```sql
create table test123 (
    id number(10,0) primary key,
    dat json
);

insert into test123(id, dat) values (1, '{}');

update test123
set dat = json_transform(dat, set '$.field1' = date '2020-01-01');

select
    json_value(dat, '$.field1' returning date),
    to_date(substr(json_value(dat, '$.field1'),1,10),'YYYY-MM-DD')
from test123;
```